app/notification_push_test.go

```diff
@@ -1362,6 +1364,48 @@ func TestAllPushNotifications(t *testing.T) {
 	assert.Equal(t, 6, numUpdateBadges)
 }
 
+func TestPushNotificationRace(t *testing.T) {
```

Comment:
Nit: A comment here that is explaining how to fail on this test would be nice to have. (like it will fail when the tests run with `-race` flag if there is any data race)

Reply:
Sure. But this test will also fail without the `-race` flag, if the code is not right.

Reply:
Hmm, we can add that to the comment too. Or would it be more elegant to recover the panic and make a `t.Fail()`?

Reply:
Hmm .. ok.